core/src/main/java/org/sonarsource/sonarlint/core/util/StringUtils.java

```diff
@@ -55,5 +60,23 @@ public static String describe(Object o) {
   public static boolean isEmpty(@Nullable String str) {
     return str == null || str.isEmpty();
   }
+  
+  public static String md5(InputStream data) throws IOException, NoSuchAlgorithmException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Refactor this method to throw at most one checked exception instead of: java.io.IOException, java.security.NoSuchAlgorithmException [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1160)